src/docfx/lib/YamlUtility.cs

```diff
@@ -151,25 +161,40 @@ private static JToken ToJson(YamlNode node)
                 {
                     if (key is YamlScalarNode scalarKey)
                     {
-                        obj[scalarKey.Value] = ToJson(value);
+                        var jToken = ToJson(value, mappings);
+                        obj[scalarKey.Value] = jToken;
                     }
                     else
                     {
                         throw new NotSupportedException($"Not Supported: {key} is not a primitive type");
                     }
                 }
+                SetMappings(mappings, node, obj);
                 return obj;
             }
             if (node is YamlSequenceNode seq)
             {
                 var arr = new JArray();
                 foreach (var item in seq)
                 {
-                    arr.Add(ToJson(item));
+                    arr.Add(ToJson(item, mappings));
                 }
+                SetMappings(mappings, node, arr);
                 return arr;
             }
             throw new NotSupportedException($"Unknown yaml node type {node.GetType()}");
         }
+
+        private static void SetMappings(Dictionary<JToken, List<(int, int)>> mappings, YamlNode scalar, JToken value)
+        {
+            if (mappings.ContainsKey(value))
```

Comment:
>ContainsKey [](start = 25, length = 11)

when will this be equal, if we use reference comparison, it should be a one one map

Reply:
Some previous e2e tests are failing while setting the mapping during parsing as below:
Message: System.ArgumentException : An item with the same key has already been added. Key: { "metadata": { "key": "value" } }